src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionBase.cs

```diff
@@ -104,6 +108,17 @@ public void MarkConnectionAsNotIdle()
             _connectionMetrics?.IdleStateChanged(idle: false);
         }
 
+        public void LinkRequestActivity()
+        {
+            Activity? requestActivity = Activity.Current;
+            if (_activity is null || requestActivity?.OperationName is not DiagnosticsHandlerLoggingStrings.RequestActivityName)
+            {
+                return;
+            }
+
+            requestActivity.AddLink(new ActivityLink(_activity.Context));
```

Comment:
Is tooling/infra prepared to deal with there being millions of links to a given connection?

Reply:
Millions of links to is not a problem, millions of links from could be!